### ISSUE TITLE
LineContinuationCheck unable to handle wrapping @return in Javadoc in Intellij 

### ISSUE THREAD

**Issue by PatrickDattilio** [checkstyle/checkstyle#3258]
Intellij has the option of aligning the `@param` description, but not the `@return`. The following gives us a LineContinuation error (on second line of `@return`).

Expected no violation, or an option to ignore in JavadocTagContinuationIndentation.

```
$ cat Test.java
package com.example.test.checkstyletest;

public class Test {

  /**
   * Intellij has options to align the @param tag description, but not the @return. Checkstyle is
   * fighting the Intellij formatter with a Javadoc
   *
   * @param - This is a really long param description. It will correctly indent due to an option in
   *          the code formatter to align the description indentation.
   * @return This is a really long return description. There is no option to align the at-return
   * so we get a JavadocTagContinuationIndentationCheck on this line.
   */
  public boolean testMethod(boolean isTrue) {
    return isTrue;
  }
}

$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocTagContinuationIndentation" />
  </module>
</module>

$ java -jar checkstyle-6.19-all.jar -c config.xml Test.java
Starting audit...
[ERROR] p:\dev\checkstyletest\Test.java:12: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
Please report an issue as described here http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by PatrickDattilio** [comment]
`$ cat Test.java
package com.example.test.checkstyletest;

public class Test {

  /**
   * Intellij has options to align the @param tag description, but not the @return. Checkstyle is
   * fighting the Intellij formatter with a Javadoc
   *
   * @param - This is a really long param description. It will correctly indent due to an option in
   *          the code formatter to align the description indentation.
   * @return This is a really long return description. There is no option to align the at-return
   * so we get a JavadocTagContinuationIndentationCheck on this line.
   */
  public boolean testMethod(boolean isTrue) {
    return isTrue;
  }
}

Pmoney@PCD /p/dev/checkstyletest
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocTagContinuationIndentation" />
  </module>
</module>
Pmoney@PCD /p/dev/checkstyletest
$ java -jar checkstyle-6.19-all.jar -c config.xml Test.java
Starting audit...
[ERROR] p:\dev\checkstyletest\Test.java:12: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 1 errors.`

**Comment by PatrickDattilio** [comment]
Expected no violation, or an option to ignore in JavadocTagContinuationIndentation.

```
$ cat Test.java
package com.example.test.checkstyletest;

public class Test {

  /**
   * Intellij has options to align the @param tag description, but not the @return. Checkstyle is
   * fighting the Intellij formatter with a Javadoc
   *
   * @param - This is a really long param description. It will correctly indent due to an option in
   *          the code formatter to align the description indentation.
   * @return This is a really long return description. There is no option to align the at-return
   * so we get a JavadocTagContinuationIndentationCheck on this line.
   */
  public boolean testMethod(boolean isTrue) {
    return isTrue;
  }
}

Pmoney@PCD /p/dev/checkstyletest
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocTagContinuationIndentation" />
  </module>
</module>
Pmoney@PCD /p/dev/checkstyletest
$ java -jar checkstyle-6.19-all.jar -c config.xml Test.java
Starting audit...
[ERROR] p:\dev\checkstyletest\Test.java:12: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
@PatrickDattilio , why you think it is problem in Checkstyle and not in IntelijIdea ?

**Comment by PatrickDattilio** [comment]
I don't think it's a _problem_ but following what you said on #3182 

> Lets follow IDEs ways. Till checkstyle can fix indentation itself (auto-fix), user will follow IDE as it do this boring job as formatting automatically.

I think it would be nice to have the ability to allow this, so we aren't fighting the automatic formatting.

**Comment by romani** [comment]
Please report this against intelij idea project to make sure that is a feature , not a bug. Please put a link to it here to keep track of it.
Checkstyle should follow features of IDEs, not a bugs.

**Comment by PatrickDattilio** [comment]
It currently seems to be somewhere between an unimplemented feature and a bug. I found this existing ticket from 2014 that has yet to be scheduled.
https://youtrack.jetbrains.com/issue/IDEA-121723
A few of us voting the issue up might not be a bad idea.

**Comment by romani** [comment]
thanks a lot for investigation!
I am closing the issue for now, as it works in Eclipse. Looks like Intelij Idea problems.
We could reopen this issue in future, as some details come from Jetbrains .
